js_modules/dagit/packages/core/src/overview/OverviewJobsRoot.tsx

```diff
@@ -33,33 +36,26 @@ export const OverviewJobsRoot = () => {
   const refreshState = useQueryRefreshAtInterval(queryResultOverview, FIFTEEN_SECONDS);
 
   // Batch up the data and bucket by repo.
-  const repoBuckets = React.useMemo(() => buildBuckets(data), [data]);
+  const repoBuckets = React.useMemo(
+    () =>
+      buildBuckets(data).filter(({repoAddress}) =>
+        visibleKeys.has(repoAddressAsString(repoAddress)),
+      ),
+    [data, visibleKeys],
```

Comment:
I don't think it's valid to use `visibleKeys` as a dep here since it's a `new Set` each render? Might need to throw a memo around `visibleRepoKeys`

Reply:
Good point. I had originally written the visible key function as a hook with a memo, I think I lost track of this in the shuffle.